### PR TITLE
[TT-11008] Extract URLRewrite

### PR DESCRIPTION
**Pull request by titpetric**
https://tyktech.atlassian.net/browse/TT-11008

### CODE REVIEW

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5949/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7555006108)

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5949/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7555006108)

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5949/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7555006108)

**Comment by buger** [comment]
**API tests result - mongo44-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5949/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7555006108)

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -355,6 +355,23 @@ type ExtendedPathsSet struct {
 	PersistGraphQL          []PersistGraphQLMeta  `bson:"persist_graphql" json:"persist_graphql"`
 }
 
+// Clear clears any value in ExtendedPathsSet. It is used in converting the values from
```
Comment (andrei-tyk) [inline apidef/api_definitions.go:358]:
```suggestion
// Clear wipes any existing values from ExtendedPathsSet. It is used in converting the values from
```

File: apidef/api_definitions.go [hunk 2]
```diff
@@ -355,6 +355,23 @@ type ExtendedPathsSet struct {
...
+// the OAS API definition, so old values don't persist when converting to classic api definition.
```
Comment (andrei-tyk) [inline apidef/api_definitions.go:359]:
```suggestion
// the OAS API definition, so old values don't persist when converting to classic API definition.
```

**Review by andrei-tyk — APPROVED** [review]

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**